lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -617,6 +617,10 @@ void LLVMToSPIRV::transVectorComputeMetadata(Function *F) {
     BF->addDecorate(DecorationSIMTCallINTEL, SIMTMode);
   }
 
+  if (Attrs.hasAttribute(AttributeList::ReturnIndex,
+                         kVCMetadata::VCSingleElementVector))
+    BF->addDecorate(DecorationSingleElementVectorINTEL);
```

Comment:
I've just realized that this implementation is not really conformant with the spec, as per extension spec this decoration should be applied to a type and not to function or its argument:

> Apply to a 'Scalar' type to indicate that the type represents a single-element vector type in the high-level language.

Reply:
Spec was updated, now the patch conforms with it